### DISCUSSION TITLE
Bug using NNCase on MacOS

### DISCUSSION THREAD

**Discussion by narduzzi** [kendryte/nncase#928] [Q&A] [answered]
Hi,

I am trying to use NNCase to convert my TFLite model for K510, but it fails.

**Describe the bug**
It seems that a library can not be loaded.

RuntimeError: Cannot load module: libnncase.targets.k510.dylib, dlopen(libnncase.targets.k510.dylib, 0x0001): tried: 'libnncase.targets.k510.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibnncase.targets.k510.dylib' (no such file), '/Users/my_user/KendryteDeployment/venv/lib/python3.9/site-packages/libnncase.targets.k510.dylib' (no such file), '/opt/anaconda3/lib/libnncase.targets.k510.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/anaconda3/lib/libnncase.targets.k510.dylib' (no such file), '/opt/anaconda3/bin/../lib/libnncase.targets.k510.dylib' (no such file), '/usr/lib/libnncase.targets.k510.dylib' (no such file, not in dyld cache), 'libnncase.targets.k510.dylib' (no such file), '/usr/local/lib/libnncase.targets.k510.dylib' (no such file), '/usr/lib/libnncase.targets.k510.dylib' (no such file, not in dyld cache)

**To Reproduce**
I have install the wheel downloaded from github

```bash
# load python3.9.12
pip install nncase-1.9.0.20230322-cp39-cp39-macosx_10_15_x86_64.whl
# works OK
```

**Expected behavior**

**Origin model and code**

```python
#!/user/bin/env python

import nncase
import os
import argparse

def read_model_file(model_file):
    with open(model_file, 'rb') as f:
        model_content = f.read()
    return model_content

def convert(filename):
    # model
    model = filename
    name = filename.split('.kmodel')[0]
    target = 'k510'

    # compile_options
    compile_options = nncase.CompileOptions()
    compile_options.target = target
    compile_options.dump_ir = True
    compile_options.dump_asm = True
    compile_options.dump_dir = 'tmp'

    # compiler
    compiler = nncase.Compiler(compile_options)

    # import_options
    import_options = nncase.ImportOptions()

    # import
    model_content = read_model_file(model)
    compiler.import_tflite(model_content, import_options)

    # compile
    compiler.compile()

    # kmodel
    kmodel = compiler.gencode_tobytes()
    with open(f'k510_{name}.kmodel', 'wb') as f:
        f.write(kmodel)

if __name__ == '__main__':
    # Arguments for reading filename
    parser = argparse.ArgumentParser(description='Convert tflite to kmodel')
    parser.add_argument('--filename', type=str, default='model.tflite', help='Filename of tflite model')
    args = parser.parse_args()

    convert(args.filename)

```

**Environment (please complete the following information):**
 - OS: MacOS 13.2.1 (22D68)
 - nncase version : nncase-1.9.0.20230322-cp39-cp39-macosx_10_15_x86_64.whl
 - DL Framework: Tensorflow 2.5.0

**Questions**
- Why is the conversion failing?
- How can I use ncc from command line?

Thanks for your help!

**Comment by zhen8838** [comment, marked as answer]
1. you need to install [nncase-k510](https://pypi.org/project/nncase-k510/), and nncase-k510 only support linux. you'd better use our docker.
2. the command line interface was deprecated.
